optimum/intel/openvino/modeling_base_seq2seq.py

```diff
@@ -257,7 +257,21 @@ def _from_pretrained(
             if not compile_only:
                 encoder = cls.load_model(file_names["encoder"], quantization_config)
                 decoder = cls.load_model(file_names["decoder"], quantization_config)
-                if use_cache:
+                if use_cache and not model_has_state(decoder):
+                    model_file_names["decoder_with_past"] = decoder_with_past_file_name
+                    model_file_names["decoder_with_past_bin"] = decoder_with_past_file_name.replace(".xml", ".bin")
```

Comment:
this should only be added and downloaded `if not from_onnx` right ?

Reply:
yes, onnx format does not support stateful models, so there is no relevant changes for from_onnx path

Reply:
my problem is that now decoder_with_past_bin will be (attempted to be) downloaded even when `from_onnx` is True.
because these binaries used to be added to `model_file_names` under the condition `if not from_onnx:` `for key in list(model_file_names.keys()):`, now that list only has encoder and decoder at that stage, to which we add `decoder_with_past` (ok) and `decoder_with_past_bin` (not necessarily exist) unconditionally.

Reply:
```suggestion
                    if not from_onnx:
                        model_file_names["decoder_with_past_bin"] = decoder_with_past_file_name.replace(".xml", ".bin")
```

Reply:
but the next loop will have to be adapted as well